language/tools/test_generation/src/abstract_state.rs

```diff
@@ -78,34 +93,48 @@ impl AbstractState {
             stack: Vec::new(),
             locals: HashMap::new(),
             register: None,
+            module: empty_module()
+                .freeze()
+                .expect("Empty module should pass the bounds checker"),
         }
     }
 
     /// Create a new AbstractState given a list of `SignatureTokens` that will be
-    /// the (available) locals that the state will have.
-    pub fn from_locals(locals: HashMap<usize, (AbstractValue, BorrowState)>) -> AbstractState {
+    /// the (available) locals that the state will have, as well as the module state
+    pub fn from_locals(
+        module: CompiledModuleMut,
+        locals: HashMap<usize, (AbstractValue, BorrowState)>,
+    ) -> AbstractState {
         AbstractState {
             stack: Vec::new(),
             locals,
             register: None,
+            module: module
+                .freeze()
+                .expect("Module should pass the bounds checker"),
```

Comment:
I'm bemused by the failure of freeze translating into knowing that a module did not pass the bounds checker. Is freeze the best name for this?

Reply:
Unfortunately I didn't pick this name so I can't say why the name is what it is.